lib/child_process.js

```diff
@@ -61,7 +62,9 @@ exports.fork = function(modulePath /*, args, options*/) {
 
   if (pos < arguments.length && arguments[pos] != null) {
     if (typeof arguments[pos] !== 'object') {
-      throw new TypeError('Incorrect value of args option');
+      throw new errors.TypeError('ERR_INVALID_OPT_VALUE',
+                                 ('arguments[' + pos + ']'),
+                                 arguments[pos]);
```

Comment:
Felt odd to me too, and still mulling over how to properly present this state to users 😞